File: app/src/main/java/com/example/projectsantander/ui/login/LoginContract.java
package com.example.projectsantander.ui;

import android.content.Context;

public class LoginContract {

        interface View {
            void showProgress(final boolean show);

            void showError(String error);

            void navigationToExtract();


            Context getContext();
        }

        interface Presenter {
            void login(String username, String password);
        }
    }

File: bankapp/app/src/main/java/dev/ornelas/bankapp/ui/login/LoginContract.kt
package dev.ornelas.bankapp.ui.login

import android.content.Intent

interface LoginContract {
    interface View {
        fun displayLoginResult(result: LoginViewModel)
    }

    interface Presenter {
        fun onLogin(username: String, password: String)
        fun onViewCreated()
    }

    interface Router {
        fun navigateToStatements(user: LoggedInUserView)
    }

}